core/src/proxy/block_engine_stage.rs

```diff
@@ -272,25 +327,39 @@ impl BlockEngineStage {
         ),
         bundle_tx: &Sender<Vec<PacketBundle>>,
         packet_tx: &Sender<PacketBatch>,
-        trust_packets: bool,
+        block_engine_config: &BlockEngineConfig,
         verified_packet_tx: &Sender<(Vec<PacketBatch>, Option<SigverifyTracerPacketStats>)>,
         exit: &Arc<AtomicBool>,
         block_builder_fee_info: &Arc<Mutex<BlockBuilderFeeInfo>>,
+        mut auth_client: AuthServiceClient<Channel>,
+        access_token: Arc<Mutex<Token>>,
+        refresh_token: &mut Token,
+        keypair: Arc<Keypair>,
+        cluster_info: &Arc<ClusterInfo>,
+        connection_timeout: &Duration,
     ) -> crate::proxy::Result<()> {
         const METRICS_TICK: Duration = Duration::from_secs(1);
         const MAINTENANCE_TICK: Duration = Duration::from_secs(10 * 60);
```

Comment:
is there a way to do this more dynamically but also simple (simple is key)? if tokens are good for 5 minutes, this logic won't work bc needs to be refreshed more frequently than MAINTENANCE_TICK

Reply:
it might make more sense to do re-auth in the 1s check

Reply:
I moved to metrics_tick.  Honestly, I had in my mind that maintenance tick was 1s.